src/utils.ts

```diff
@@ -0,0 +1,32 @@
+import {IScore} from "./types";
+
+const SCIENCE_KEYS = [
+	'compass',
+	'tablet',
+	'gear'
+] as Array<keyof IScore>;
+
+export function getSum(playerScore: IScore): number {
+	return getFlatSum(playerScore) + getScienceSum(playerScore);
+}
+
+export function getFlatSum(playerScore: IScore): number {
+	return (Object.keys(playerScore) as Array<keyof IScore>).reduce((sum, key) => {
```

Comment:
Object.keys(playerScore).filter(k=>!SCIENCE_KEYS.includes(k)).reduce((sum,key)=>sum+playerScore[key], 0);

Reply:
Time complexity is a little bit worse with two walks instead of the only walk

Reply:
But allocation rate on each iteration is much higher.